bitbots_localization/src/localization.cpp

```diff
@@ -466,266 +466,62 @@ void Localization::publish_particle_markers() {
-  marker.pose.orientation.w = 1.0;
-  marker.type = visualization_msgs::Marker::POINTS;
-  marker.scale.x = 0.2;
-  marker.scale.y = 0.2;
-  marker.lifetime = ros::Duration(1.0);
-
-  for (std::pair<double, double> &corner : corners) {
-    //lines are in polar form!
-    std::pair<double, double> cornerRelative;
-
-    cornerRelative = corner_->observationRelative(corner, best_estimate.getXPos(), best_estimate.getYPos(),
-                                                  best_estimate.getTheta());
-    double occupancy = corner_->get_occupancy(cornerRelative.first, cornerRelative.second);
-    geometry_msgs::Point point;
-    point.x = cornerRelative.first;
-    point.y = cornerRelative.second;
-    point.z = 0;
-
-    std_msgs::ColorRGBA color;
-    color.b = 1;
-    if (occupancy >= 0) {
-      color.r = occupancy / 100;
-    } else {
-      color.g = 1;
-    }
-    color.a = 1;
-    marker.points.push_back(point);
-    marker.colors.push_back(color);
-  }
-  corner_ratings_publisher_.publish(marker);
-}
-
-void Localization::publish_t_crossings_ratings() {
-  std::vector<std::pair<double, double>> tcrossings = robot_pose_observation_model_->get_measurement_t_crossings();
-  RobotState best_estimate = robot_pf_->getBestXPercentEstimate(config_.percentage_best_particles);
-  double rating = 0;
-
-  visualization_msgs::Marker marker;
-  marker.header.frame_id = "map";
-  marker.header.stamp = ros::Time::now();
-  marker.ns = "tcrossings_ratings";
-  marker.action = visualization_msgs::Marker::ADD;
-  marker.pose.orientation.w = 1.0;
-  marker.type = visualization_msgs::Marker::POINTS;
-  marker.scale.x = 0.2;
-  marker.scale.y = 0.2;
-  marker.lifetime = ros::Duration(1.0);
-
-  for (std::pair<double, double> &tcrossing : tcrossings) {
-    //lines are in polar form!
-    std::pair<double, double> tcrossingRelative;
-
-    tcrossingRelative = t_crossings_map_->observationRelative(tcrossing, best_estimate.getXPos(),
-                                                              best_estimate.getYPos(),
-                                                              best_estimate.getTheta());
-    double occupancy = t_crossings_map_->get_occupancy(tcrossingRelative.first, tcrossingRelative.second);
-    geometry_msgs::Point point;
-    point.x = tcrossingRelative.first;
-    point.y = tcrossingRelative.second;
-    point.z = 0;
-
-    std_msgs::ColorRGBA color;
-    color.b = 1;
-    if (occupancy >= 0) {
-      color.r = occupancy / 100;
-    } else {
-      color.g = 1;
-    }
-    color.a = 1;
-    marker.points.push_back(point);
-    marker.colors.push_back(color);
-  }
-  t_crossings_ratings_publisher_.publish(marker);
-}
-
-void Localization::publish_crosses_ratings() {
-  std::vector<std::pair<double, double>> corners = robot_pose_observation_model_->get_measurement_crosses();
-  RobotState best_estimate = robot_pf_->getBestXPercentEstimate(config_.percentage_best_particles);
+  RobotState best_estimate = robot_pf_->getBestState();
```

Comment:
Why getBestState? If I remember corectly, this returns the one best state. But I think the more expected behaviour would be to do the visualization based on the published pose estimate.

Reply:
I agree with you, but until now the line_ratings used the best state, so left it unchanged